src/onnx/onnx.cpp

```diff
@@ -705,11 +740,13 @@ struct onnx_parser
     parse_pooling(const std::string& name, node_info info, std::vector<instruction_ref> args)
     {
         op::pooling op{ends_with(name, "MaxPool") ? "max" : "average"};
-        auto l0 = args[0];
+        auto l0      = args[0];
+        auto in_lens = l0->get_shape().lens();
+        auto kdims   = in_lens.size() - 2;
```

Comment:
Should we check that `in_lens.size()` is greater than 2?

Reply:
fixed